### PR TITLE
[go-sdk] Add docs to match the TypeScript SDK

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
Adds Go SDK docs that mirror the TypeScript docs.

Some parts are commented out that should be there, but the functionality doesn't exist.

### Checklist

- Do all Lints pass?
  - [x] Have you ran `pnpm spellcheck`?
  - [x] Have you ran `pnpm fmt`?
  - [x] Have you ran `pnpm lint`?

### CODE REVIEW

File: apps/nextra/pages/en/build/sdks/go-sdk.mdx [hunk 1]
```diff
@@ -2,16 +2,10 @@
 title: "Go SDK"
 ---
 
-import { Callout } from "nextra/components";
```
Comment (hariria) [inline apps/nextra/pages/en/build/sdks/go-sdk.mdx:5]:
```suggestion
import { Cards, Card } From "@components/index
```

**Review by hariria — CHANGES_REQUESTED** [review]
Nextra is failing to build due to missing `Card` import on `apps/nextra/pages/en/build/sdks/go-sdk.mdx`

File: apps/nextra/pages/en/build/sdks/go-sdk.mdx (hunk 1 above)
Comment (gregnazario) [reply apps/nextra/pages/en/build/sdks/go-sdk.mdx:5]: Fixed

File: apps/nextra/pages/en/build/sdks/go-sdk.mdx [hunk 2]
```diff
@@ -20,78 +16,20 @@ To use the Go SDK, get the main package here:
...
-
-	// Wait for the transaction
-	_, err = client.WaitForTransaction(hash)
-
-  // Read transaction by hash
-	txn, err := client.TransactionByHash(hash)
-```
-
-### More examples
+## Navigation
```
Comment (hariria) [inline apps/nextra/pages/en/build/sdks/go-sdk.mdx:19]: Maybe we can call this Usage instead

File: apps/nextra/pages/en/build/sdks/go-sdk.mdx [hunk 3]
```diff
@@ -20,78 +16,20 @@ To use the Go SDK, get the main package here:
...
+
+<Cards>
+  <Card href="./go-sdk/fetch-data-via-sdk.mdx">
```
Comment (hariria) [inline apps/nextra/pages/en/build/sdks/go-sdk.mdx:22]: This link seems to be broken on preview. I think you need to omit the .mdx

File: apps/nextra/pages/en/build/sdks/go-sdk.mdx [hunk 4]
```diff
@@ -20,78 +16,20 @@ To use the Go SDK, get the main package here:
...
+    <Card.Title>Fetching Data</Card.Title>
+    <Card.Description>Learn how to fetch data with the Go SDK</Card.Description>
+  </Card>
+  <Card href="./go-sdk/building-transactions.mdx">
```
Comment (hariria) [inline apps/nextra/pages/en/build/sdks/go-sdk.mdx:26]: Link is broken

File: apps/nextra/pages/en/build/sdks/go-sdk.mdx [hunk 5]
```diff
@@ -20,78 +16,20 @@ To use the Go SDK, get the main package here:
...
+    <Card.Title>Submitting Transactions</Card.Title>
+    <Card.Description>Learn how to submit transactions with the Go SDK</Card.Description>
+  </Card>
+  <Card href="./go-sdk/go-examples.mdx">
```
Comment (hariria) [inline apps/nextra/pages/en/build/sdks/go-sdk.mdx:30]: Link is broken

File: apps/nextra/pages/en/build/sdks/go-sdk/building-transactions.mdx [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+  Finally, you can wait for the result of the transaction by using
+  `client.WaitForTransaction()` and specifying the hash of the transaction you
+  just submitted like so:
+
+  ```go filename="build_a_transaction.go"
+  // 5. Wait for the transaction to complete
+  txnHash := submitResult.Hash
+  _, err = client.WaitForTransaction(txnHash)
+  ```
+  <Fragment></Fragment>
```
Comment (hariria) [inline apps/nextra/pages/en/build/sdks/go-sdk/building-transactions.mdx:136]: Is the fragment needed?

File: apps/nextra/pages/en/build/sdks/go-sdk/building-transactions/multi-agent-transactions.mdx [hunk 7]
```diff
@@ -0,0 +1,92 @@
+import { Callout, Steps } from 'nextra/components'
```
Comment (hariria) [inline apps/nextra/pages/en/build/sdks/go-sdk/building-transactions/multi-agent-transactions.mdx:1]: Add title frontmatter

File: apps/nextra/pages/en/build/sdks/go-sdk/building-transactions/batching-transactions.mdx [hunk 8]
```diff
@@ -0,0 +1,142 @@
+# Batching Transactions
```
Comment (hariria) [inline apps/nextra/pages/en/build/sdks/go-sdk/building-transactions/batching-transactions.mdx:1]: Add title frontmatter

File: apps/nextra/pages/en/build/sdks/go-sdk/building-transactions.mdx [hunk 9]
```diff
@@ -0,0 +1,328 @@
+import { Callout, Steps } from 'nextra/components';
```
Comment (hariria) [inline apps/nextra/pages/en/build/sdks/go-sdk/building-transactions.mdx:1]: Add title frontmatter

File: apps/nextra/pages/en/build/sdks/go-sdk/fetch-data-via-sdk.mdx [hunk 10]
```diff
@@ -0,0 +1,101 @@
+import { Callout } from 'nextra/components'
```
Comment (hariria) [inline apps/nextra/pages/en/build/sdks/go-sdk/fetch-data-via-sdk.mdx:1]: Add title frontmatter

**Review by hariria — APPROVED** [review]
Major thing is the links are broken on the go sdk landing page

File: apps/nextra/pages/en/build/sdks/go-sdk/building-transactions.mdx (hunk 6 above)
Comment (gregnazario) [reply apps/nextra/pages/en/build/sdks/go-sdk/building-transactions.mdx:136]: I don't know... this was randomly in the TS SDK one I copied

**Pull request merged**